theano/misc/pycuda_example.py

```diff
@@ -285,6 +285,11 @@ def __init__(self, scalar_op, inplace_pattern=None, name=None):
         self.scalar_op = scalar_op
         self.inplace_pattern = inplace_pattern
 
+    # As we have a dict in props, we need to implement __hash__
+    def __hash__(self):
+        return (hash(type(self)) ^ hash(self.scalar_op) ^
+                hash_from_dict(self.inplace_pattern))
```

Comment:
that isn't always safe as some int have collision....

On Sat, Aug 8, 2015 at 6:25 PM, Pascal Lamblin <notifications@github.com>
wrote:

> In theano/misc/pycuda_example.py
> <https://github.com/Theano/Theano/pull/3263#discussion_r36582483>:
>
> > @@ -285,6 +285,11 @@ def __init__(self, scalar_op, inplace_pattern=None, name=None):
> >          self.scalar_op = scalar_op
> >          self.inplace_pattern = inplace_pattern
> >
> > +    # As we have a dict in props, we need to implement __hash__
> > +    def __hash__(self):
> > +        return (hash(type(self)) ^ hash(self.scalar_op) ^
> > +                hash_from_dict(self.inplace_pattern))
>
> Is that the right way? Or should the default __hash__ method be updated
> to call hash_from_dict instead?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Theano/Theano/pull/3263/files#r36582483>.
>